tests/TritiumAndEfficiencyBinner_test.py

```diff
@@ -0,0 +1,88 @@
+
+
+        specGen_config = {
+            "volume": 7e-6*1e-2, # [m3]
+            "density": 3e17, # [1/m3]
+            "duration": 1.*seconds_per_year()/12., # [s]
+            "neutrino_mass" :0, # [eV]
+            "energy_window": [tritium_endpoint()-1e3,tritium_endpoint()+1e3], # [KEmin,KEmax]
+            "frequency_window": [-100e6, +100e6], #[Fmin, Fmax]
+            "energy_or_frequency": "frequency",
+            "background": 0,#1e-6, # [counts/eV/s]
+            "energy_resolution": 5,# [eV]
+            "frequency_resolution": 2e6,# [Hz]
+            "mode": "generate",
+            "varName": "F",
+            "iter": 10000,
+            "interestParams": "F",
+            "fixedParams": {"m_nu": 0},
+            "options": {"snr_efficiency": True, "channel_efficiency":False, "smearing": False},
+            "snr_efficiency_coefficients": [-265.03357206889626, 6.693200670990694e-07, -5.795611253664308e-16, 1.5928835520798478e-25, 2.892234977030861e-35, -1.566210147698845e-44],
+            "channel_central_frequency": 1400e6,
+            "mixing_frequency": 24.5e9
+        }
+        tritiumAndEfficiencyBinner_config = {
+            "energy_or_frequency": 'frequency', #Currently only set up to use frequency
+            "variables": "F",
+            'bins': np.linspace(24.5e9+1300e6, 24.5e9+1550e6, 25),
+            'fss_bins': False, # If fss_bins is True, bins is ignored and overwritten
+            'efficiency_filepath': 'combined_energy_corrected_eff_at_quad_trap_frequencies.json'
+        }
+
+
+        specGen = DistortedTritiumSpectrumLikelihoodSampler("specGen")
+        tritiumAndEfficiencyBinner = TritiumAndEfficiencyBinner("tritiumAndEfficiencyBinner")
+
+        specGen.Configure(specGen_config)
+        tritiumAndEfficiencyBinner.Configure(tritiumAndEfficiencyBinner_config)
+
+
+        specGen.Run()
+        data = specGen.data
+
+        tritiumAndEfficiencyBinner.data = data
+        tritiumAndEfficiencyBinner.Run()
+        results = tritiumAndEfficiencyBinner.results
+
+        plt.figure()
+        plt.subplot(1,2,1)
+        plt.errorbar(results['F'], results['N'], yerr = np.sqrt(results['N']), drawstyle = 'steps-mid')
+        plt.xlabel('Frequency [Hz]')
+        plt.ylabel('Counts')
+        plt.subplot(1,2,2)
+        plt.errorbar(results['F'], results['bin_efficiencies'], yerr = results['bin_efficiency_errors'], drawstyle = 'steps-mid')
+        plt.xlabel('Frequency [Hz]')
+        plt.ylabel('Bin efficiency')
+        plt.tight_layout()
+        plt.savefig('TritiumAndEfficiencyBinnerOutputPlot.png')
+
+        plt.figure()
+        plt.errorbar(data['F'], results['event_efficiencies'], yerr = results['event_efficiency_errors'], fmt = 'none')
+        plt.xlabel('Frequency [Hz]')
+        plt.ylabel('Event efficiency')
+        plt.tight_layout()
+        plt.savefig('TritiumAndEfficiencyBinnerOutputPlotByEvent.png')
+
+if __name__ == '__main__':
```

Comment:
Since morpho v2.7.1, it is possible to add the following paragraph here:
```python
    args = parser.parse_args(False)
    logger = morphologging.getLogger('morpho',
                                     level=args.verbosity,
                                     stderr_lb=args.stderr_verbosity,
                                     propagate=False)
    logger = morphologging.getLogger(__name__,
                                     level=args.verbosity,
                                     stderr_lb=args.stderr_verbosity,
                                     propagate=False)
```
in order to propagate morpho commands into the script. 
This propagation includes the verbosity controls; as it is right now, your script would print only info level logs.
With this new paragraph, a `-v` in the `python TritiumAndEfficiencyBinner_test.py -v` will allow you to also have debug logs.

Reply:
A good example can be found here: https://github.com/morphoorg/morpho/pull/147/files#diff-ac68cdaeb26b6040380c08bc8d2047c1

Reply:
Is this worth doing even for processors that do not have debug statements, like the TritiumAndEfficiencyBinner?

Reply:
Right, it is indeed not necessary for test scripts that don't have processors with debug statements.It seems that TritiumAndEfficiencyBinner contains MultiCicada processor which seems to have one. but again, this is not mandatory, just nicer :)

Reply:
Nice, thank you.

Reply:
Hi Mathieu – When @cclaessens and I add this test but don't include -v during execution, we both see debug statements printed during running. This is not supposed to happen, right?

Reply:
@guiguem, more specifically, here is what I am seeing:

When the -v option is included, all logger statements print (I think - there may be some that can only be seen for a higher verbosity setting, and I do not know about them).

When the -v option is not included, I still see debug statements from the BaseProcessor. I do not see any info statements. When I do an analysis with the PyStanSamplingProcessor and CalibrationProcessor, the Stan result table is not printed, and no output at all is provided from the CalibrationProcessor.

Reply:
Indeed this is not supposed to happen; I am not sure what is going on.
I guess this doesn't prevent the release to happen; we should just open an issue and deal wit this later...

Reply:
Sounds good.

To slightly amend my statement above, when I run without -v, I do see one info statement at the very beginning:
`2020-06-04T18:47:00[INFO    ] PhylloxeraPy(96) -> All set!`
But I see no info statements after that.